x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/azureaistudio/completion/AzureAiStudioChatCompletionServiceSettingsTests.java

```diff
@@ -119,4 +123,38 @@ public void testToFilteredXContent_WritesAllValues() throws IOException {
     public static HashMap<String, Object> createRequestSettingsMap(String target, String provider, String endpointType) {
         return new HashMap<>(Map.of(TARGET_FIELD, target, PROVIDER_FIELD, provider, ENDPOINT_TYPE_FIELD, endpointType));
     }
+
+    @Override
+    protected Writeable.Reader<AzureAiStudioChatCompletionServiceSettings> instanceReader() {
+        return AzureAiStudioChatCompletionServiceSettings::new;
+    }
+
+    @Override
+    protected AzureAiStudioChatCompletionServiceSettings createTestInstance() {
+        return createRandom();
+    }
+
+    @Override
+    protected AzureAiStudioChatCompletionServiceSettings mutateInstance(AzureAiStudioChatCompletionServiceSettings instance)
+        throws IOException {
+        return randomValueOtherThan(instance, AzureAiStudioChatCompletionServiceSettingsTests::createRandom);
+    }
+
+    @Override
+    protected AzureAiStudioChatCompletionServiceSettings mutateInstanceForVersion(
+        AzureAiStudioChatCompletionServiceSettings instance,
+        TransportVersion version
+    ) {
+        return instance;
```

Comment:
Why not implement like the above `mutateInstance`?

Reply:
As we don't have anything to mutate yet for any specific transport version for `AzureAiStudioChatCompletionServiceSettings` I don't think we should mutate something. This acts more like a guard, if we do BWC incompatible changes in the future to implement it. That's at least my understanding of `mutateInstanceForVersion`